apps/server/src/modules/tool/service/validation/tool-validation.service.ts

```diff
@@ -35,16 +35,17 @@ export class ToolValidationService {
 
 		const loadedTool: ExternalToolDO = await this.externalToolService.findExternalToolById(toolId);
 		if (
-			loadedTool.config instanceof Oauth2ToolConfigDO &&
+			this.externalToolService.isOauth2Config(loadedTool.config) &&
```

Comment:
what happens if this resolves to false? then the type is not checked, and its possible to overwrite the data is it not?

I would at least like to see a test for this case